test/mpfr.jl

```diff
@@ -1,898 +1,1017 @@
+    f = BigFloat(10)^Int32(1000)
+    @test i1 != i2
+    @test i1 != f
+    @test i2 != f
+
+    @test f > i1
+    @test f > i2
+
+    i3 = trunc(Integer,f)
+    @test i3 == f
+    @test i3+1 > f
+    @test i3+1 >= f
+
+    let err(z, x) = abs(z - x) / abs(x)
+        @test 1e-60 > err(eta(parse(BigFloat,"1.005")), parse(BigFloat,"0.693945708117842473436705502427198307157819636785324430166786"))
+        @test 1e-60 > err(exp(eta(big(1.0))), 2.0)
+    end
+end
 
-# PR 17217 -- BigFloat constructors with given precision and rounding mode
+@testset "issue #8318" begin
+    @test convert(Int64, big(500_000_000_000_000.)) == 500_000_000_000_000
+end
 
-# test constructors and `big` with additional precision and rounding mode:
+@testset "issue #9816" begin
+    # check exponent range is set to max possible"
+    @test MPFR.get_emin() == MPFR.get_emin_min()
+    @test MPFR.get_emax() == MPFR.get_emax_max()
+end
 
-for prec in (10, 100, 1000)
-    for val in ("3.1", pi, "-1.3", 3.1)
-        let
-            a = BigFloat(val)
-            b = BigFloat(val, prec)
-            c = BigFloat(val, RoundUp)
-            d = BigFloat(val, prec, RoundDown)
-            e = BigFloat(val, prec, RoundUp)
+@testset "issue #10994: handle embedded NUL chars for string parsing" begin
+    @test_throws ArgumentError parse(BigFloat, "1\0")
+end
+
+@testset "serialization (issue #12386)" begin
+    let b = IOBuffer()
+        x = 2.1*big(pi)
+        serialize(b, x)
+        seekstart(b)
+        @test deserialize(b) == x
+    end
+
+    @test isnan(sqrt(BigFloat(NaN)))
+end
 
-            @test precision(a) == precision(BigFloat)
-            @test precision(b) == prec
-            @test precision(c) == precision(BigFloat)
-            @test precision(d) == prec
-            @test precision(e) == prec
-            (val != 3.1) && @test e > d     # rounding has no effect when constructing from Float64
+@testset "PR 17217: BigFloat constructors with given precision and rounding mode" begin
+
```

Comment:
Not a big deal but this line break probably isn't necessary. Stylistically it differs from the rest of the file.